src/sm.c

```diff
@@ -333,14 +334,14 @@ SM_RESULT sm_exec_begin(SM_HANDLE sm)
         else
         {
             /*Codes_SRS_SM_02_056: [ sm_exec_begin shall increment n. ]*/
-            (void)InterlockedIncrement(&sm->non_barrier_call_count);
-            LONG state2 = InterlockedAdd(&sm->state, 0);
+            (void)interlocked_increment(&sm->non_barrier_call_count);
+            LONG state2 = interlocked_add(&sm->state, 0);
 
             /*Codes_SRS_SM_02_057: [ If the state changed after incrementing n then sm_exec_begin shall return SM_EXEC_REFUSED. ]*/
             if (state1 != state2)
             {
                 LogError("sm name=%s. state changed meanwhile from %" PRI_SM_STATE " to %" PRI_SM_STATE "", sm->name, SM_STATE_VALUE(state1), SM_STATE_VALUE(state2));
-                LONG n = InterlockedDecrement(&sm->non_barrier_call_count);
+                LONG n = interlocked_decrement(&sm->non_barrier_call_count);
                 if (n == 0)
                 {
                     WakeByAddressSingle((void*)&sm->non_barrier_call_count);
```

Comment:
>WakeByAddressSingle [](start = 20, length = 19)

wake_by_address_single (while we're changing the rest of this)?

Reply:
Will do a separate PR, I want to not leave InterlockedHL hanging in the nether

---
In reply to: [501166830](https://github.com/Azure/azure-c-util/pull/66#discussion_r501166830) [](ancestors = 501166830)